src/buildstream/sandbox/_sandboxbuildboxrun.py

```diff
@@ -91,8 +114,10 @@ def _execute_action(self, action, flags):
         casd = cascache.get_casd()
         config = self._get_config()
 
-        if config.remote_apis_socket_path and context.remote_cache_spec:
-            raise SandboxError("'remote-apis-socket' is not currently supported with 'storage-service'.")
+        if config.remote_apis_socket_path and context.remote_cache_spec and not self.re_remote:
+            raise SandboxError(
+                "'remote-apis-socket' is not supported with 'storage-service' without 'action-cache-service'."
```

Comment:
So IIUC, we now support

* fully local: the buildbox-casd started by buildstream handles everything (storage, execution, action cache)
* fully remote: the remote execution servers handle everything (cache storage-service could double as storage for remote execution)
* local with remote action cache: execution is done locally, storage can be either the `cache` storage-service or the `remote-execution` storage-service, action cache has to be configured in `remote-execution`.

Assuming the above is correct, I find that this error message a bit confusing as you can't set the `action-cache-service` in the `cache` configuration.

Reply:
I've adjusted the error message, hope it's clearer now.